### DISCUSSION TITLE
The magic actions $refresh does not work

### DISCUSSION THREAD

**Discussion by takeshiyu** [livewire/livewire#3424] [1. Help]
I'm working a lot with Livewire on a new project. After upgrading to v2.5.5, the magic actions `$refresh` does not work any more. For example:

```php
<?php

namespace App;

use Illuminate\Support\Facades\Storage;

/**
 * ShowOrder
 */
class ShowOrder extends \Livewire\Component
{
    use \Livewire\WithFileUploads;

    protected $listeners = ['refresh' => '$refresh'];

    public function uploadImages()
    {
        $this->uploadToS3();

        $this->emit('hideModal');
        $this->emit('refresh');       
    }
}
```
The `refresh` event does not fire. Anything that I am missing?

**Comment by joshhanley** [comment]
@s12i can I ask why you are calling refresh on the same component?

That just adds an unnecessary network request, as the `uploadImages` request would finish, then the component does another request to refresh.
